File: front-end/src/app/domain/models/game/board/location/locationImpl/TaxLocation.ts
import {Location} from "../Location";

export class TaxLocation extends Location {
  private _name: string;
  private _amount: number;

  public constructor(
    id: number,
    position: number,
    description: string,
    name: string,
    amount: number
  ) {
    super(TaxLocation.name, id, position, description);
    this._name = name;
    this._amount = amount;
  }

  public static trueCopy(location: TaxLocation): TaxLocation | null {
    return new TaxLocation(
      location.id,
      location.position,
      location.description,
      location.name,
      location.amount,
    );
  }

  public override toJson(): object {
    return Object.assign(
      super.toJson(),
      {
        "name": this.name,
        "amount": this.amount,
      }
    );
  }

  public get name(): string {
    return this._name;
  }

  public get amount(): number {
    return this._amount;
  }

  public override toString(): string {
    return this.name;
  }

}

File: back-end/src/main/java/nl/hva/backend/domain/models/game/board/location/locationImpl/TaxLocation.java
package nl.hva.backend.domain.models.game.board.location.locationImpl;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nl.hva.backend.domain.models.game.board.GameBoard;
import nl.hva.backend.domain.models.game.board.location.Location;

import javax.persistence.Entity;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class TaxLocation extends Location {
    private String name;
    private double amount;

    public TaxLocation(
            long id,
            int position,
            String description,
            GameBoard gameBoard,
            String name,
            double amount
    ) {
        super(id, position, description, gameBoard);
        this.name = name;
        this.amount = amount;
    }
}